### PR TITLE
Update RAJA submodule and CI

### PR DESCRIPTION
**Pull request by rhornung67**
# Summary 

- This PR updates the RAJA submodule to the develop branch and also updates both GItLab and azure CI to match what RAJA does, with a few differences.

### CODE REVIEW

**Comment by rhornung67** [comment]
@adrienbernede I tried a new PR that is basically the same as yours but uses the same Spack version as RAJA. 

The clang13 and clang14 failures on lassen seem to only be reproducible in the CI. They are always an OpenMP variant of the Apps_HALOEXCHANGE_FUSED kernel. They have been passing in other PRs that use the CI stuff in RAJAPerf develop and the code for those kernels hasn't changed. So something may have changed in RAJA between v2022.10.5 (what RAJAPerf develop is using) and v2023.06.0 (latest release, used here). Will keep digging.

**Comment by adrienbernede** [comment]
Great! I indeed forgot to update Spack.

**Comment by rhornung67** [comment]
> Great! I indeed forgot to update Spack.

@adrienbernede the issue with seg faults in the CI for clang13 and clang14 on lassen in GitLab CI is not understood. The segfaults occur in the Base_OpenMP or Lambda_OpenMP variants, which do not use RAJA. They will pass eventually if I restart the CI jobs. I have run these builds manually several times on rzansel and have not been able to reproduce the seg faults, for example
`SPEC=" ~shared +openmp %clang@13.0.1 cflags==\"-DGTEST_HAS_CXXABI_H_=0\" cxxflags==\"-stdlib=libc++ -DGTEST_HAS_CXXABI_H_=0\"" lalloc 1 -W 30 ./scripts/gitlab/build_and_test.sh`

At this point, I'm thinking that there is an issue with the OpenMP compilation or runtime since the failures are intermittent. I'm going to try a couple more things on this PR.

**Review by artv3 — APPROVED** [review]

File: src/apps/HALOEXCHANGE_FUSED-OMP.cpp [hunk 1]
```diff
@@ -91,7 +92,8 @@ void HALOEXCHANGE_FUSED::runOpenMPVariant(VariantID vid, size_t RAJAPERF_UNUSED_
             buffer += len;
           }
         }
-#if _OPENMP >= 200805
+
+#if defined(RAJA_ENABLE_OMP_TASK_INTERNAL)
```
Comment (artv3) [inline src/apps/HALOEXCHANGE_FUSED-OMP.cpp:96]: Nice!

Comment (rhornung67) [reply src/apps/HALOEXCHANGE_FUSED-OMP.cpp:96]: There is an exposed CMake option for this 'RAJA_ENABLE_OMP_TASK' in RAJA and a corresponding Spack variant in the RAJA and RAJAPerf spack packages. The result is that we can enable/disable internal RAJA algorithm variants for things like OpenMP sort for different builds/tests based on compiler.  We didn't see the need for this before, but something is fishy with newer clang compilers on blueos systems.

File: src/apps/HALOEXCHANGE_FUSED-OMP.cpp [hunk 2]
```diff
@@ -148,7 +150,8 @@ void HALOEXCHANGE_FUSED::runOpenMPVariant(VariantID vid, size_t RAJAPERF_UNUSED_
             buffer += len;
           }
         }
-#if _OPENMP >= 200805
+#if 0
+//#if _OPENMP >= 200805
```
Comment (MrBurmark) [inline src/apps/HALOEXCHANGE_FUSED-OMP.cpp:154]:
```suggestion
#if defined(RAJA_ENABLE_OMP_TASK_INTERNAL)
```

File: src/apps/HALOEXCHANGE_FUSED-OMP.cpp [hunk 3]
```diff
@@ -186,7 +189,8 @@ void HALOEXCHANGE_FUSED::runOpenMPVariant(VariantID vid, size_t RAJAPERF_UNUSED_
...
```
Comment (MrBurmark) [inline src/apps/HALOEXCHANGE_FUSED-OMP.cpp:193]:
```suggestion
#if defined(RAJA_ENABLE_OMP_TASK_INTERNAL)
```

File: src/apps/HALOEXCHANGE_FUSED-OMP.cpp (hunk 2 above)
Comment (rhornung67) [reply src/apps/HALOEXCHANGE_FUSED-OMP.cpp]: Thanks. Forgot to clean up my mess.

File: src/apps/HALOEXCHANGE_FUSED-OMP.cpp (hunk 3 above)
Comment (rhornung67) [reply src/apps/HALOEXCHANGE_FUSED-OMP.cpp]: Fixed.

**Review by MrBurmark — APPROVED** [review]

**Pull request merged**